Sources/Main.cpp

```diff
@@ -1268,5 +1280,126 @@ void __fastcall TX584Form::AboutItemClick(TObject *Sender)
 {
     AboutForm->ShowModal();
 }
+
+//---------------------------------------------------------------------------
+//                           *** БУФЕР ОБМЕНА ***
+//---------------------------------------------------------------------------
+
+int TX584Form::GetClipboardBufferSize()
+{
+    int result = 0;
+
+    result += 2; // количество элементов в буфере
+    for (int i = 0; i < ClipboardSize; i++) {
+		result += 2 + 1 + CMClipboard[i].Length() * 2;
+    }
+
+    return result;
+}
+//---------------------------------------------------------------------------
+
+void TX584Form::PutIntoClipboard()
+{
+    HGLOBAL hBuffer = NULL;
+    unsigned char *Buffer, *p;
+    int BufferSize;
+
+    Clipboard()->Open();
+    Clipboard()->Clear();
+    BufferSize = GetClipboardBufferSize();
+
+    hBuffer = GlobalAlloc(GMEM_MOVEABLE, BufferSize);
+    Buffer = (unsigned char *)GlobalLock(hBuffer);
+    p = Buffer;
+
+    // записать количество элементов в буфере обмена
```

Comment:
Raw memory manipulations are hard to read and error-prone. Use `TMemoryStream` + `TBinaryWriter` instead of manually constructing the clipboard contents. This is much simpler and safer way to work with arbitrary binary data. For example, both microinstruction and comment can we written to memory stream in a couple of lines:

```cpp
Writer->Write(MIClipboard[i]);
Writer->Write(CMClipboard[i]);
```

Make sure to specify UTF-8 encoding while creating `TBinaryWriter`. This allows lossless conversion of Unicode strings to raw binary data.

Remove `GetClipboardBufferSize` method. Use `Memory` and `Size` properties of `TMemoryStream` to get access to the underlying memory buffer and copy it to the clipboard.